tests/batchcheck/batchcheck.go

```diff
@@ -0,0 +1,209 @@
+	name := test.Name
+
+	if contextTupleTest {
+		name += "_ctxTuples"
+	}
+
+	t.Run(name, func(t *testing.T) {
+		if contextTupleTest && len(test.Stages) > 1 {
+			// we don't want to run special contextual tuples test for these cases
+			// as multi-stages test has expectation tuples are in system
+			t.Skipf("multi-stages test has expectation tuples are in system")
+		}
+
+		t.Parallel()
+		resp, err := client.CreateStore(ctx, &openfgav1.CreateStoreRequest{Name: name})
+		require.NoError(t, err)
+
+		storeID := resp.GetId()
+
+		for stageNumber, stage := range test.Stages {
+			t.Run(fmt.Sprintf("stage_%d", stageNumber), func(t *testing.T) {
+				if contextTupleTest && len(stage.Tuples) > 20 {
+					// https://github.com/openfga/api/blob/05de9d8be3ee12fa4e796b92dbdd4bbbf87107f2/openfga/v1/openfga.proto#L151
+					t.Skipf("cannot send more than 20 contextual tuples in one request")
+				}
+				// arrange: write model
+				writeModelResponse, err := writeAuthModel(ctx, client, storeID, stage.Model)
+				require.NoError(t, err)
+
+				if !contextTupleTest {
+					err = writeTuples(ctx, client, storeID, stage.Tuples, writeModelResponse)
+					require.NoError(t, err)
+				}
+
+				if len(stage.CheckAssertions) == 0 {
+					t.Skipf("no check assertions defined")
+				}
+
+				// map of correlation_id to result
+				expectedResults := map[string]*batchTestResult{}
+
+				// checks to be passed into batch check request
+				protoChecks := make([]*openfgav1.BatchCheckItem, 0, len(stage.CheckAssertions))
+
+				for _, assertion := range stage.CheckAssertions {
+					// monkey patch the contextual tuples since we don't actually define them in yaml
+					if contextTupleTest {
+						assertion.ContextualTuples = append(assertion.ContextualTuples, stage.Tuples...)
+					}
+
+					correlationID := ulid.Make().String()
+
+					item := batchchecktest.BatchCheckItemFromCheckAssertion(assertion, correlationID)
+					protoChecks = append(protoChecks, item)
+					expectedResults[correlationID] = &batchTestResult{
+						Allowed:   assertion.Expectation,
+						ErrorCode: assertion.ErrorCode,
+					}
+				}
+
+				resp, err := client.BatchCheck(ctx, &openfgav1.BatchCheckRequest{
+					StoreId:              storeID,
+					AuthorizationModelId: writeModelResponse.GetAuthorizationModelId(),
+					Checks:               protoChecks,
+				})
+				require.NoError(t, err)
+
+				result := resp.GetResult()
+
+				for correlationID, expected := range expectedResults {
+					thisResult := result[correlationID]
+					if expected.ErrorCode == 0 {
+						require.Equal(t, expected.Allowed, thisResult.GetAllowed())
+						require.Nil(t, thisResult.GetError())
+						continue
+					}
+
+					testErr := thisResult.GetError()
+					inputErrorCode := testErr.GetInputError().Number()
+					if inputErrorCode != 0 {
+						require.Equal(t, expected.ErrorCode, int(inputErrorCode))
+					} else {
+						require.Equal(t, expected.ErrorCode, int(testErr.GetInternalError().Number()))
+					}
```

Comment:
Not a big deal but if you want to get rid of the `continue` 🤷‍♂️ 

```suggestion
					if expected.ErrorCode != 0 {
					        testErr := thisResult.GetError()
					        inputErrorCode := testErr.GetInputError().Number()
					        if inputErrorCode != 0 {
						        require.Equal(t, expected.ErrorCode, int(inputErrorCode))
					        } else {
						        require.Equal(t, expected.ErrorCode, int(testErr.GetInternalError().Number()))
					        }
					}
					require.Equal(t, expected.Allowed, thisResult.GetAllowed())
					require.Nil(t, thisResult.GetError())

```

Reply:
I actually don't know the go community's thoughts on `continue` in general, but I'll leave it cuz the increased indentation bothers my dev ocd